sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py

```diff
@@ -97,15 +97,15 @@ def _build_connection_policy(kwargs):
         policy.SSLConfiguration = ssl
 
     # Retry config
-    retry = kwargs.pop('retry_options', None) or policy.RetryOptions
```

Comment:
This kwarg was supported for backwards compatibility (these are passed straight through from the client constructor). We can't remove it completely, but we could also add a deprecation warning to that one.
If we also wanted to avoid supporting these legacy options in the async client - I would be okay with that.

Reply:
added a deprecation warning along with putting the kwarg back. thank you again for the feedback.